weldx/core.py

```diff
@@ -0,0 +1,492 @@
+
+        """
+        return len(self._parameters)
+
+    @property
+    def num_variables(self):
+        """Get the expressions number of free variables.
+
+        Returns
+        -------
+        int:
+            Number of free variables.
+
+        """
+        return len(self.expression.free_symbols) - len(self._parameters)
+
+    @property
+    def expression(self):
+        """Return the internal sympy expression.
+
+        Returns
+        -------
+        Internal sympy expression
+
+        """
+        return self._expression
+
+    @property
+    def parameters(self) -> Dict:
+        """Return the internal parameters dictionary.
+
+        Returns
+        -------
+        Dict
+            Internal parameters dictionary
+
+        """
+        return self._parameters
+
+    def get_variable_names(self):
+        """Get a list of all expression variables.
+
+        Returns
+        -------
+        List
+
+        """
+        variable_names = []
+        for var in self._expression.free_symbols:
+            if var.__str__() not in self._parameters:
+                variable_names.append(var.__str__())
+        return variable_names
+
+    def evaluate(self, **kwargs) -> Any:
+        """Evaluate the expression for specific variable values.
+
+        Parameters
+        ----------
+        kwargs
+            additional keyword arguments (variable assignment) to pass.
+
+        Returns
+        -------
+        Any:
+            Result of the evaluated function
+
+        """
+        intersection = set(kwargs).intersection(self._parameters)
+        if len(intersection) > 0:
+            raise ValueError(
+                f"The variables {intersection} are already defined as parameters."
+            )
+        inputs = {**kwargs, **self._parameters}
+        return self.function(**inputs)
+
+
+# TimeSeries ---------------------------------------------------------------------------
+
+
+class TimeSeries:
+    """Describes the behaviour of a quantity in time."""
+
+    _valid_interpolations = ["step", "linear"]
+
+    def __init__(
+        self,
+        data: Union[pint.Quantity, MathematicalExpression],
+        time: Union[None, pd.TimedeltaIndex] = None,
+        interpolation: str = "linear",
+    ):
+        """Construct a TimSeries.
+
+        Parameters
+        ----------
+        data:
+            Either a pint.Quantity or a weldx.MathematicalExpression. If a mathematical
+            expression is chosen, it is only allowed to have a single free variable,
+            which represents time.
+        time:
+            An instance of pandas.TimedeltaIndex if a quantity is passed and 'None'
+            otherwise.
+        interpolation:
+            A string defining the desired interpolation method. This is only relevant if
+            a quantity is passed as data. Currently supported interpolation methods are:
+            'step', 'linear'.
+
+        """
+        self._time = None
```

Comment:
I think we can remove this and just `return None` by default at the end of `@property def time(...`

Reply:
Done